src/ActionParameterHandler.ts

```diff
@@ -103,32 +103,30 @@ export class ActionParameterHandler {
     /**
      * Normalizes parameter value.
      */
-    protected normalizeParamValue(value: any, param: ParamMetadata): Promise<any>|any {
+    protected async normalizeParamValue(value: any, param: ParamMetadata): Promise<any> {
         if (value === null || value === undefined)
             return value;
 
+        // map @QueryParams object properties from string to basic types (normalize)
+        if (param.type === "queries" && typeof value === "object") {
+            Object.keys(value).map(key => {
+                const type = Reflect.getMetadata("design:type", param.targetType.prototype, key);
+                const typeString = typeof type();
```

Comment:
TBH, I don't like this part, I did this as a quick shortcut.
It may introduce some weird behavior when the reflected type is not constructor-like, however I'm not sure if it's possible in TypeScript? I think I should check if the `type` is undefined for case like no property type provided in class definition.